llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -7892,6 +7925,12 @@ bool CombinerHelper::matchShuffleDisjointMask(MachineInstr &MI,
   if (getOpcodeDef(TargetOpcode::G_IMPLICIT_DEF, Shuffle.getSrc2Reg(), MRI))
     return false;
 
+  if (getOpcodeDef(TargetOpcode::G_POISON, Shuffle.getSrc1Reg(), MRI))
+    return false;
```

Comment:
In the combine patch, we should use something smarter for matching. I don't particularly like getOpcodeDef, and this should probably use the mi_match on an UndefValue similar to how the IR does it